### ISSUE TITLE
[Enhancement] Jakarta EE9+ without classifiers

### ISSUE THREAD

**Issue by papegaaij** [apache/shiro#1351]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/shiro/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

At the moment Shiro uses the `jakarta` classifier for jars that have been transformed to the Jakarta EE namespace. This makes it hard to get the dependencies correct, because these jars share their dependencies with the non-Jakarta variants.

### Describe the solution you'd like

Many projects have moved their source code to Jakarta EE 9 or higher. Shiro could start version 3 with Jakarta EE 10. It should not be that hard to keep supporting both 2.x and 3.x for some time, as the code is mostly identical.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by lprimak** [comment]
That's being discussed currently on the mailing list.
Another topic is to keep base Java version as 11 or take it to 17 or 21.
Your opinion would be welcome.

**Comment by papegaaij** [comment]
Yes, I know. I'm the one who started the discussion. I would require Java 17. This will be required for Jakarta EE 11 as well. It has good support both in tools and servers. It would make sense to test on 21 as well, as it's the next LTS and servers like WildFly will make this the recommended SE version soon (probably WildFly 32 or 33).

**Comment by adityavikramsinha** [comment]
Came across this issue in one of my projects while migrating to `jakarta.*`. It appears a lot of the shiro-web classes (among others) are still using `javax.*`. There is a work around given [here](https://github.com/apache/shiro/issues/1324#issuecomment-2083168422). 
I was thinking of manually patching it by forking. I am not on the mailing list so I am unsure of the conversation regarding this issue and its potential standard fixes.

**Comment by lprimak** [comment]
@adityavikramsinha Are you using Shiro with OSGi or Karaf?

**Comment by adityavikramsinha** [comment]
@lprimak No, I’m using Shiro in a standard Spring Boot application (not OSGi or Karaf)

**Comment by lprimak** [comment]
In that case, the issue should not apply to you.
If you use the BOM and correct classifiers (see example https://github.com/apache/shiro/tree/main/samples/spring-boot-3-web) you should have no problems.

**Comment by adityavikramsinha** [comment]
Thankyou for this, it worked!!
